src/coreclr/binder/inc/assembly.hpp

```diff
@@ -27,90 +27,38 @@
 
 #include "bundle.h"
 
-STDAPI BinderAcquirePEImage(LPCTSTR            szAssemblyPath,
-                            PEImage          **ppPEImage,
-                            BundleFileLocation bundleFileLocation);
-
-STDAPI BinderAcquireImport(PEImage            *pPEImage,
-                           IMDInternalImport **pIMetaDataAssemblyImport,
-                           DWORD              *pdwPAFlags);
-
-STDAPI BinderReleasePEImage(PEImage *pPEImage);
-
-STDAPI BinderAddRefPEImage(PEImage *pPEImage);
-
 namespace BINDER_SPACE
 {
-
-    // An assembly represents a particular set of bits.  However we extend this to
-    // also include whether those bits have precompiled information (NGEN).   Thus
-    // and assembly knows whether it has an NGEN image or not.
-    //
-    // This allows us to preferentially use the NGEN image if it is available.
+    // BINDER_SPACE::Assembly represents a result of binding to an actual assembly (PEImage)
+    // It is basically a tuple of 1) physical assembly and 2) binder which created/owns this binding
+    // We also store whether it was bound using TPA list
     class Assembly
```

Comment:
Yes. I think this needs to be renamed, but could not yet come up with a good name. 
Mechanically speaking it could be `BoundPEImage`, but I do not think `PEImage` is itself a very intuitive name. Besides we are binding assemblies, not images. 

The image is basically an assembly loaded into the process, but not yet "loaded" by the Loader (that would be a PEAssembly). That is the distinction between `PEImage` and `PEFile`. They are not much related to each other, or to files. One is "loaded" by the Loader and another is input for the Loader.

We could benefit from a better naming, but better naming is not easy :-)
Also maybe having fewer things called Assembly could help too. We also have `DomainAssembly`, for example.

Reply:
I did not get into details of `BindResult` yet. I assume it can represent a trial/failing attempt to bind with some context for the failure, but that is just a guess as I think we would need something like that and `BindResult` is that.

I will definitely look at what can be shared or whether we need a separate type.